mubench.pipeline/data/detector.py

```diff
@@ -1,33 +1,38 @@
 from os.path import join, exists
-from typing import Optional, List
+from typing import Dict, Optional, List
 
 from data.finding import Finding, SpecializedFinding
-
+from data.runner_interface import RunnerInterface
+from utils.io import read_yaml
 
 class Detector:
     BASE_URL = "http://www.st.informatik.tu-darmstadt.de/artifacts/mubench/detectors"
-    CLI_VERSION = "v20170406"
+    RELEASES_FILE = "releases.yml"
 
     def __init__(self, detectors_path: str, detector_id: str, java_options: List[str]):
         self.id = detector_id
         self.base_name = detector_id.split("_", 1)[0]
-        self.java_options = java_options
-
         self.path = join(detectors_path, self.id)
-        self.jar_path = join(self.path, self.base_name + ".jar")
-        self.jar_url = "{}/{}/{}.jar".format(Detector.BASE_URL, Detector.CLI_VERSION, self.base_name)
-        self.md5_path = join(self.path, self.base_name + ".md5")
 
-    @property
-    def md5(self) -> Optional[str]:
-        md5_file = self.md5_path
-        md5 = None
+        releases_index_path = join(self.path, Detector.RELEASES_FILE)
+        release_info = self._get_release_info(releases_index_path)
+        self.release_tag = release_info.get("tag", "latest")
+        self.md5 = release_info.get("md5", None)
+        self.cli_version = release_info.get("cli_version", None)
 
-        if exists(md5_file):
-            with open(md5_file) as file:
-                md5 = file.read().strip()
-
-        return md5
+        self.jar_path = join(self.path, self.base_name + ".jar")
+        self.jar_url = "{}/{}/{}.jar".format(Detector.BASE_URL, self.cli_version, self.base_name)
```

Comment:
We need to include the tag in the URL. I'll move the files on the server accordingly.